### PR TITLE
Add a guide about ZNC

### LINKED ISSUE #149: Write page "Set up ZNC with The Lounge"
Would live at http://thelounge.chat/docs/guides/znc

Point out that you might need to use ZNC at all.

### PR DESCRIPTION
**Pull request by brunnre8**
Closes #149.

First draft of how a znc guide could look like.
May you please have a look and tell me what needs to be improved?

### CODE REVIEW

File: _guides/znc.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+- Attach multiple clients (say a computer, phone, tablet etc.) to the bouncer, all sharing the same nick.
+
+- Messages are saved even when no client is connected, playing back what was said when a client comes back
+
+- As the bouncer (preferably) never disconnects, mode changes, nick names and channels are kept at all times
+
+- Sharing block lists / using additional functionality across different clients, without having to configure each one of them
+
+## What thelounge gives you out of the box
+Thelounge already gives you most of the features you expect from a bouncer:
```
Comment (Zarthus) [inline _guides/znc.md]: The Lounge (brand name)

Comment (Zarthus) [reply _guides/znc.md]: also everywhere else it's used

Comment (brunnre8) [reply _guides/znc.md]: yeah, just wanted to mention that I'll fix it everywhere

**Comment by brunnre8** [comment]
Alternative layout as per @Zarthus suggestion

```
## What is ZNC?

[...]

## What does it do that The Lounge doesn't?

[...]

## I'm convinced, how do I set it up?

[install]

### Connecting to ZNC using The Lounge

[connecting]
```

File: _guides/znc.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.
+
+In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:
+
+ - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. Works with any client even when they are not znc aware
+
+ - [Playback](https://wiki.znc.in/Playback): IRC clients may request the module to send a partial buffer playback starting from and ending to a certain point of time. Only works with clients that respect the znc specific capability. The Lounge supports this functionality out of the box
+
+## Connect to multiple Networks from The Lounge
+For each IRC server connected from ZNC, you will need a separate irc server connection from within The Lounge.
```
Comment (xPaw) [inline _guides/znc.md]:
```suggestion
For each IRC server connected in ZNC, you will need a separate IRC server connection in The Lounge.
```

File: _guides/znc.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+- Detach from channels and auto-attach on activity
+
+- Run multiple The Lounge versions, sharing the data (for example stable and dev in order to test the next The Lounge PR)
+
+## Setting up ZNC
+If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.
+
+In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:
+
+ - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. Works with any client even when they are not znc aware
```
Comment (xPaw) [inline _guides/znc.md]:
```suggestion
 - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. This works with any client even when they are not ZNC aware
```

File: _guides/znc.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+---
+
+## What The Lounge gives you out of the box
+The Lounge already gives you most of the features you expect from a bouncer:
+
+- Scroll back is already available from within The Lounge, even across restarts, if logging to sqlite is enabled
+
+- Multiple devices can connect to The Lounge at the same time, using The Lounge as a client
+
+- Nick, mode changes etc. are kept, as The Lounge never disconnects from the IRC server
```
Comment (xPaw) [inline _guides/znc.md]:
```suggestion
- The Lounge keeps a persistent connection to the IRC server even if you don't have the client open anywhere.
```

File: _guides/znc.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+- Push notifications are available on Android and on modern browsers
```
Comment (xPaw) [inline _guides/znc.md]:
```suggestion
- Push notifications are available if they are supported by your platform and browser
```

File: _guides/znc.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+---
+layout: documentation
+title: Set up ZNC with The Lounge
+---
+
+## What The Lounge gives you out of the box
+The Lounge already gives you most of the features you expect from a bouncer:
+
+- Scroll back is already available from within The Lounge, even across restarts, if logging to sqlite is enabled
```
Comment (xPaw) [inline _guides/znc.md]:
```suggestion
- Scrollback is already available from within The Lounge, even across restarts, [if message storage is configured](/docs/configuration#messagestorage)
```

File: _guides/znc.md [hunk 7]
```diff
@@ -0,0 +1,38 @@
+---
+layout: documentation
+title: Set up ZNC with The Lounge
+---
+
+## Why you may not need znc
```
Comment (xPaw) [inline _guides/znc.md]:
```suggestion
## Why you may not need ZNC
```

File: _guides/znc.md [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+## Setting up ZNC
+If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.
+
+In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:
+
+ - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. Works with any client even when they are not znc aware
+
+ - [Playback](https://wiki.znc.in/Playback): IRC clients may request the module to send a partial buffer playback starting from and ending to a certain point of time. Only works with clients that respect the znc specific capability. The Lounge supports this functionality out of the box
+
+## Connect to multiple Networks from The Lounge
```
Comment (xPaw) [inline _guides/znc.md]:
```suggestion
## Connect to multiple networks
```

File: _guides/znc.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+- Multiple devices can connect to The Lounge at the same time, using The Lounge as a client.
+
+- The Lounge keeps a persistent connection to the IRC server even if you don't have the client open anywhere.
+
+- Push notifications are available if they are supported by your platform and browser.
+
+## Setting up ZNC
+If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.
+
+In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:
```
Comment (astorije) [inline _guides/znc.md]:
```suggestion
In order for The Lounge to properly integrate with ZNC (and other clients), you may want to use either one of the following modules:
```

File: _guides/znc.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+
+- Scrollback is already available from within The Lounge, even across restarts, [if message storage is configured](/docs/configuration#messagestorage).
+
+- Multiple devices can connect to The Lounge at the same time, using The Lounge as a client.
+
+- The Lounge keeps a persistent connection to the IRC server even if you don't have the client open anywhere.
+
+- Push notifications are available if they are supported by your platform and browser.
+
+## Setting up ZNC
```
Comment (astorije) [inline _guides/znc.md]:
```suggestion
## Set up ZNC
```

To be consistent with the next bullet (and most pages of the website, if I'm correct).

File: _guides/znc.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+If you still like to have the additional benefits ZNC gives you, install ZNC according to the [installation instructions](https://wiki.ZNC.in/Installation) for your system.
+
+In order for The Lounge to properly integrate with The Lounge (and other clients) you probably want either one of the following modules:
+
+ - [Clientbuffer](https://wiki.znc.in/Clientbuffer): The client buffer module maintains client specific buffers for identified clients. This works with any client even when they are not ZNC aware.
+
+ - [Playback](https://wiki.znc.in/Playback): IRC clients may request the module to send a partial buffer playback starting from and ending to a certain point of time. Only works with clients that respect the ZNC specific capability. The Lounge supports this functionality out of the box.
+
+## Connect to multiple networks
+For each IRC server connected in ZNC, you will need a separate IRC server connection in The Lounge.
```
Comment (astorije) [inline _guides/znc.md]: > connected in ZNC

Is it just me or "in" sounds off?
Would something else seem better? "through", "via", "over", etc.

File: _guides/znc.md [hunk 12]
```diff
@@ -0,0 +1,40 @@
+---
+layout: documentation
+title: Set up ZNC with The Lounge
+description: How to configure ZNC for use with The Lounge
```
Comment (astorije) [inline _guides/znc.md]:
```suggestion
description: Extend the server capabilities of The Lounge using ZNC and its modules
```

File: _guides/znc.md (hunk 9 above)
Comment (brunnre8) [reply _guides/znc.md]: I'm going to replace the first 'The Lounge' instance, else the "(and other clients)" doesn't make any sense. Is that ok?

File: _guides/znc.md (hunk 11 above)
Comment (brunnre8) [reply _guides/znc.md]: That was a change request from @xPaw. The original was "from The Lounge" I believe.
What do you prefer?

Comment (Zarthus) [reply _guides/znc.md]: connected through ZNC

**Comment by astorije** [comment]
Thank you so much for your help with this, @brunnre8! As you can see, your guide is now available at https://thelounge.chat/docs/guides
Hope you [requested your stickers](https://goo.gl/forms/f5usqAEp5DWoeXC92), you truly deserve them :)

**Pull request merged**